packages/synthetix-main/contracts/modules/core/MarketCollateralModule.sol

```diff
@@ -0,0 +1,104 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@synthetixio/core-contracts/contracts/ownership/OwnableMixin.sol";
+import "../../interfaces/IMarketCollateralModule.sol";
+import "../../storage/CollateralStorage.sol";
+import "../../storage/MarketManagerStorage.sol";
+
+import "../../utils/ERC20Helper.sol";
+
+contract MarketCollateralModule is IMarketCollateralModule, CollateralStorage, MarketManagerStorage, OwnableMixin {
+    using ERC20Helper for address;
+
+    error InsufficientMarketCollateralDepositable(uint marketId, address collateralType, uint amountToDeposit);
+    error InsufficientMarketCollateralWithdrawable(uint marketId, address collateralType, uint amountToWithdraw);
```

Comment:
i've been moving errors and events into the interface; not sure what the standard is here but felt cleaner